app/controllers/spree/adyen_redirect_controller.rb

```diff
@@ -21,6 +21,16 @@ def confirm
       end
     end
 
+    def authorise3d
+      payment = Spree::Payment.find_by(number: params[:payment_reference])
+      order = payment.order
+      payment_method = payment.payment_method
+      payment.request_env = request.env
+      payment_method.perform_authorization_3d(payment, adyen_3d_params)
+      order.next!
```

Comment:
We'll need to be careful with this. Some stores allow editing the fields in previous states without moving the order back through the state machine. So anyone who edited their payment method and hit this endpoint might error due to `confirm` not having a `next` transition.